### DISCUSSION TITLE
integration of filamentcolors library

### DISCUSSION THREAD

**Discussion by kinghat** [Donkie/Spoolman#372] [Ideas]
https://filamentcolors.xyz/library/ has an extensive library of filament pictures and data that could be incorporated.

**Comment by cnf** [comment]
I came here to suggest the exact same thing! I think it would be awesome if we can import/link our filament to the library.

**Comment by Donkie** [comment]
Very cool website, didn't know it existed!

Not sure how exactly to incorporate it though, any suggestions? It doesn't have data on the things we're mostly interested about, diameter and density...

**Reply by cnf** [reply]
That's a good question, colors/RAL etc import maybe? a lookup for vendor?

Maybe it would be possible to request they add the data we need, and have a bit of a collaboration?

I'll drop them a link to this thread on mastodon, or something, see what comes of it

**Reply by Donkie** [reply]
A collaboration would be cool indeed. I have started on a database project recently: https://github.com/Donkie/SpoolmanDB , perhaps this data could be synchronized into a single place.

**Reply by cnf** [reply]
I dropped them a poke on mastodon https://ohai.social/@cnf/112446483472723835 we'll see where we end up.

**Reply by kinghat** [reply]
the filament images could be brought into the db.

**Comment by itsthejoker** [comment]
Hi!!! I'm the filament librarian dude, thanks to @cnf for the poke. I actually have found Spoolman before, but wasn't sure if there was a way to integrate the data so I didn't reach out 😅 I definitely don't want to cause more work for @kinghat, so I encourage that an integration only be pursued if it actually fits the software rather than shoehorning in something for the cool points 😂.

If you do want to integrate it somehow, I do provide a full and open API for the website here: https://filamentcolors.xyz/api/ — and if you need specific data, let me know and I'm happy to either help you get that information or write an endpoint specific to your use cases.

**Reply by Donkie** [reply]
Hi! You have a lot of good data, but as I mentioned above, the primary need for Spoolman is the physical properties of the filament such as weight, diameter and density, which understandably is not something of particular interest for your database since it's about the colors. But do you perhaps still have information like this stored but not exposing it?

**Reply by itsthejoker** [reply]
Oh god I pinged the wrong person 😭 

I do not measure or store this information anywhere, sorry. Wouldn't this change a lot from spool to spool?

**Reply by kinghat** [reply]
wouldnt pulling in color data and images to the spoolman db be useful? i have colors that are in storage that i dont quite remember what they look like. having those images would be nice. also the hueforge integration would be useful for people who use that.

**Reply by Donkie** [reply]
Alright :)
It's not an exact science but filament measurement is surprisingly accurate enough to estimate the remaining filament for most purposes, given that you have the right values for things like density, diameter etc.

**Reply by Donkie** [reply]
> wouldnt pulling in color data and images to the spoolman db be useful? i have colors that are in storage that i dont quite remember what they look like. having those images would be nice. also the hueforge integration would be useful for people who use that.

Yes that could be interesting, but is not something I'm going to pursue right now. Hex color is good enough for most purposes in my opinion.

**Comment by runtpacket** [comment]
Loving this. @Donkie  would it be possible for us to opt into sharing our filaments/vendors from our spoolman interface? i.e "do you want to share with community" then anything you add gets merged?

**Reply by Donkie** [reply]
While this sounds good in theory, it would involve quite a lot of content moderation work, which I want to minimize. The advantage of doing it the current way with SpoolmanDB is that it requires a bar of effort for the contributor.

**Comment by SNR-Tech-Bytes** [comment]
Sorry to wake an old thread. I'm fooling around a bit with this as I have around 100 spools and am slowly adding them to spoolman (finally). My current workflow is make manufacturer, then make the spool, then make the filament. When I'm making the filament I grab the info from filamentcolors before adding the density and diameter. The one thing that would greatly accelerate this is integrate spool names with filmaentcolors so all of the manufacturers (or pretty darn close to all of them) exist in the DB. (Example, just found cookiecad doesnt exist in your current db, added manually.) Since spool weight is optional I typically leave it. Additionally having color would be awesome. When you go to make filaments I go to filamentcolors to grab the exact hex code for the filament so I know its right. I think you could *maybe* have a baked in interface connection in filaments with filament colors such that you type in the name you get a similar drop down to that of the manufacturers entry in filaments, then click the name and it auto populates the color field with the hex code from filamentcolors. It should feel pretty natural to the current interface and you'll still have to put density and diameter in manually but it at least reduces workflow for name and colors. (Filamentcolors naming scheme is really quite nice so I try to copy it anyway). I'm a pretty bad programmer but I might try to take a look at the hooks to see if I can help with that in any way
